### PR TITLE
Enable users api endpoints /users and /users/<id>

### LINKED ISSUE #3670: Implement api endpoints for `users` model
Parent issue #3459
To be done after #3645

### PR DESCRIPTION
**Pull request by shubham-padia**
Fixes #3670.

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

- `GET /v1/users` - Get users - should support filtering, sorting and pagination [check this once if it fits the desired requirements ]
- `POST /v1/users` - Create user
- `GET /v1/users/<user_id>` - Get a single user
- `PATCH /v1/users/<user_id>` - Update a single user
- `DELETE /v1/users/<user_id>` - Soft Delete a single user
- Login is required to access any endpoint, but once a user is logged in, the user can meddle with any user's info (what a wonderful wonderful world). Unfortunately, Permissions need to be discussed first in a seperate issue, are we going to implement something like client id, etc which are out of scope of this PR.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 <!--
 (Thanks for sending a pull request! Please make sure you click the link above to view the contribution guidelines, then fill out the blanks below.)
 -->
+<!-- Add the issue number that is fixed by this PR (In the form Fixes #123) -->
+
+Fixes #
```
Comment (shubham-padia) [inline .github/PULL_REQUEST_TEMPLATE.md]: Automatic issue closing was not working somehow, that's why this has been done

File: app/models/user.py [hunk 2]
```diff
@@ -68,6 +69,46 @@ class User(db.Model):
...
+    def email(self, email):
+        """
+        Setter for _email, can be only set once
+        :param email:
+        :return:
+        """
+        if self._email is None:
+            self._email = email
+        else:
+            raise AttributeError("Email cannot be modified")
```
Comment (shubham-padia) [inline app/models/user.py]: Travis is failing  due to the hybrid properties. Don't know the reason. Looking into it.

Comment (shubham-padia) [inline app/models/user.py]: @niranjan94 What should be done here. fb and google auth tests are failing due to hybrid properties (Don't know why, keeping only one of email and password hybrid also causes the error). Should the email and password check be reverted for now , or what else should be done ?

**Comment by niranjan94** [comment]
Remove the fb and google auth test. They anyway need to be re implemented.

**Pull request merged**